feature_engine/scaling/mean_normalization.py

```diff
@@ -0,0 +1,186 @@
+)
+from feature_engine._docstrings.init_parameters.all_trasnformers import (
+    _variables_numerical_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_not_learn_docstring,
+    _fit_transform_docstring,
+    _inverse_transform_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+    inverse_transform=_inverse_transform_docstring,
+)
+class MeanNormalizationScaler(BaseNumericalTransformer):
+    """
+    The MeanNormalizationScaler() applies the mean normalization scaling techinques
+    to one or multuple columns of a dataframe.
+
+    Mean normalization is a way to implement feature scaling. Mean normalization
+    calculates and subtracts the mean of for every feature and divide this value
+    by the range (max - min).
+
+    A list of variables can be passed as an argument. Alternatively, the transformer
+    will automatically select and transform all variables of type numeric.
+
+    If a column is constant, then it will be transformed to the zero constant column.
+
+    Parameters
+    ----------
+    {variables}
+
+
+    Attributes
+    ----------
+    params_:
+        a dictionary containing the mean, max and min of every given variable
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {inverse_transform}
+
+    transform:
+        Scale the variables using mean normalization.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> from feature_engine.scaling import MeanNormalizationScaler
+    >>> np.random.seed(42)
+    >>> X = pd.DataFrame(dict(x = np.random.lognormal(size = 100)))
+    >>> mns = MeanNormalizationScaler()
+    >>> mns.fit(X)
+    >>> X = mns.transform(X)
+    >>> X.head()
+            x
+    0  0.496714
+    1 -0.138264
+    2  0.647689
+    3  1.523030
+    4 -0.234153
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
+
+        self.variables = _check_variables_input_value(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        The method populate the variable `params_`.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+        self.params_ = {
```

Comment:
this is OK. I think I still prefer having one dict for mean and one dict for range. Otherwise the dict might get too big and it might get confusing when the data has too many variables. Sklearn has 2 attributes, 1 for mean or center and one for scale or range, we would keep it consistent with their api as well if we create 2 attributes.

Reply:
Ok so let's do `self.mean_` and `self.range_`